Individuals sanctions plugin fixes + workflow observability  (2 commits)

alonp99 opened the pull request:
<img width="2748" alt="Screenshot 2025-03-08 at 22 06 26" src="https://github.com/user-attachments/assets/23ed8ddd-bd1f-41b5-877e-63afe23ad207" />
<img width="2139" alt="Screenshot 2025-03-08 at 22 06 19" src="https://github.com/user-attachments/assets/3999412b-f102-47e3-8949-a7ded2e0a18d" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a “Fill Placeholders” button in the collection flow, allowing users to auto-populate context fields.
  - Enhanced workflow logs display with an interactive graph view and a dedicated modal, and added a Logs button in the workflow table for easy access to detailed log information.
  - Added new endpoints for retrieving and summarizing workflow logs, along with improved logging capabilities across various services.
  - Implemented a new `WorkflowLogController` to manage workflow log operations, including fetching logs and log types.
  
- **Style**
  - Updated the current step indicator’s color for improved visual clarity.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alonp99 committed:
feat(workflows): add workflow logging feature

- Introduce new WorkflowLog model and associated types
- Implement WorkflowLog repository, service, and controller
- Modify environment configuration to include workflow logging settings

alonp99 committed:
fix(workflows-plugins): individuals sanaction plugin fixes

tomer-shvadron commented on apps/kyb-app/src/common/components/atoms/StepperProgress/StepperProgress.tsx:
```diff
@@ -13,7 +13,7 @@ export const StepperProgress = ({ currentStep, totalSteps }: Props) => {
         <span className="text-s font-medium text-slate-600">{t('step')}</span>
         <div className="flex items-center gap-2">
-          <span className="text-s font-medium text-blue-500">{currentStep}</span>
+          <span className="text-s font-medium text-blue-600">{currentStep}</span>
```
Is this change on purpose?

alonp99 replied:
Yeah. Nitzan asked for it

tomer-shvadron commented on apps/kyb-app/src/pages/CollectionFlow/versions/v1/CollectionFlowV1.tsx:
```diff
@@ -303,6 +303,149 @@ export const CollectionFlowV1 = withSessionProtected(() => {
+                                              );
+
+                                              stateApi.setContext(filledPayload);
+                                              console.log('Context updated successfully');
```
remove all `console.log`s

tomer-shvadron commented on apps/kyb-app/src/pages/CollectionFlow/versions/v2/CollectionFlowV2.tsx:
```diff
@@ -247,6 +247,155 @@ export const CollectionFlowV2 = withSessionProtected(() => {
+                                              console.log('Filling placeholders...');
+
+                                              const filledPayload = { ...stateApi.getContext() };
+                                              console.log('Current context:', filledPayload);
```
remove all `console.log`s

tomer-shvadron commented on packages/common/src/schemas/documents/default-context-schema.ts:
```diff
@@ -21,6 +21,21 @@ export const defaultInputContextSchema = Type.Object({
   documents: DocumentsSchema,
 });
 
+export const defaultPluginSchema = Type.Object({
```
Don't forget to run `pnpm update-packages`

tomer-shvadron commented on services/workflows-service/src/workflow/workflow.controller.external.ts:
```diff
@@ -553,15 +553,17 @@ export class WorkflowControllerExternal {
-          workflowRuntime.projectId,
-          transaction,
-        );
+        if (params.event && params.event !== 'undefined') {
```
When is `params.event` equal to `'undefined'`?

alonp99 replied:
In some cases, when the unified API doesn't get an event in the callback, it will send an undefined

alonp99 committed:
fix(workflows-plugins): individuals sanaction plugin fixes

alonp99 committed:
fix(workflows-plugins): individuals sanaction plugin fixes

alonp99 committed:
test(workflow): add WorkflowLogService to various integration tests

- Import WorkflowLogService in integration test files for consistency
- Include WorkflowLogService in the providers for dependency injection in tests
- Ensure WorkflowLogService is referenced in all relevant test setups

alonp99 committed:
fix(workflows-plugins): individuals sanaction plugin fixes

alonp99 committed:
test(useRuleEngine): enhance test case for delayed execution handling

- Add waitFor to ensure state updates are applied before assertion
- Improve readability and reliability of the test by using waitFor

alonp99 committed:
fix(CollectionFlow): remove unnecessary console logs for cleaner code

- Remove debug console logs related to filling placeholders
- Enhance readability by eliminating commented-out code
